src/auth/auth.ts

```diff
@@ -735,6 +737,28 @@ export class BaseAuth<T extends AbstractAuthRequestHandler> {
         return decodedIdToken;
       });
   }
+
+  /**
+   * Enable or disable ID token verification. This is used to safely short-circuit token verification with the
+   * Auth emulator. When disabled ONLY unsigned tokens will pass verification, production tokens will not pass.
+   * 
+   * WARNING: This is a dangerous method that will compromise your app's security and break your app in
+   * production. Developers should never call this method, it is for internal testing use only.
+   * 
+   * @internal
+   */
+  // @ts-expect-error: this method appears unused but is used privately.
+  private setJwtVerificationEnabled(enabled: boolean): void {
+    if (!enabled && !useEmulator()) {
+      // We only allow verification to be disabled in conjunction with
```

Comment:
Consider throwing instead of silently failing.

Reply:
I figured silently failing adds a little security-by-obscurity since we don't actually want to help people enable this.  So if they randomly call it in an inappropriate environment, we just ignore them rather than throwing an error that may help them call it.

I realize that's not real security, so I can go either way.  @yuchenshi ?

Reply:
I'd agree with @hiranya911 here and I think it should just throw, so doing it in production will consistently fail. With that being said, the error message doesn't need to be helpful since it's an uncommon case. Something like `this method is only available in the emulators` can point them at the right direction.